plugins/modules/one_image.py

```diff
@@ -356,26 +412,32 @@ def run(self, one, module, result):
         enabled = params.get('enabled')
         new_name = params.get('new_name')
         persistent = params.get('persistent')
+        create = params.get('create')
+        template = params.get('template')
+        datastore_id = params.get('datastore_id')
+        wait_timeout = params.get('wait_timeout')
 
         self.result = {}
 
         image = self.get_image_instance(id, name)
         if not image and desired_state != 'absent':
+            if create:
+                self.result = self.create_image(name, template, datastore_id, wait_timeout)
```

Comment:
Shouldn't the module fail if `id is not None`?

Reply:
I might musunderstood you but if i got it right then it is restricted by `required_if` and `mutually_exclusive` so if you would set `create: true` then you must set a `name` but it is mutually exclusve to an ID so there are should be no problem, no?
```python
        required_if = [
            ['state', 'renamed', ['id']],
            ['create', True, ['template', 'datastore_id', 'name']],
        ]
        mutually_exclusive = [
            ['id', 'name'],
        ]
```

Reply:
Ah, I missed the `mutually_exclusive`. Thanks!